### PR TITLE
Add the regenerate-bundle-batch API endpoint

### PR DESCRIPTION
**Pull request by mprahl**
This allows a user to submit multiple regenerate-bundle requests as part of a single batch.

Additionally, this fixes the OpenAPI spec for regenerate-bundle and changes messaging.py to use the correct logger.

### CODE REVIEW

File: iib/web/messaging.py [hunk 1]
```diff
@@ -164,53 +238,46 @@ def send_message_for_state_change(request, new_batch_msg=False):
...
+    envelopes = []
+
+    for request in requests:
+        request_envelope = _get_request_state_change_envelope(request)
+        if request_envelope:
+            envelopes.append(request_envelope)
+        else:
+            # If no request envelope was generated, then that means IIB is not configured to
+            # send request state change messages and the rest of the loop can be skipped
+            break
```
Comment (lcarva) [inline iib/web/messaging.py:274]: This looks like a minor optimization that may cause some unexpected errors in the future due to unrelated code changes.

File: iib/web/messaging.py [hunk 2]
```diff
@@ -164,53 +238,46 @@ def send_message_for_state_change(request, new_batch_msg=False):
...
+
+    # Just use the last request's batch since the batch is the same for all of them
+    batch = request.batch
```
Comment (lcarva) [inline iib/web/messaging.py:277]: I find that using a variable, that has been defined in a for-loop statement, outside the for-loop to increase the difficulty in understanding and debugging the code. Consider taking the batch ID as a function parameter. You could even consider raising an error if any of the requests have a different batch ID.

File: iib/web/messaging.py [hunk 3]
```diff
@@ -164,53 +238,46 @@ def send_message_for_state_change(request, new_batch_msg=False):
...
+
+def send_messages_for_new_batch_of_requests(requests):
+    """
+    Send the appropriate message(s) based on a new batch of build requests.
+
+    If IIB is not configured to send messages, this function will do nothing.
+
+    :param list requests: the requests that were created as part of the batch request
+    """
+    if not requests:
```
Comment (lcarva) [inline iib/web/messaging.py:262]: Missing unit test coverage here.

File: iib/web/messaging.py (hunk 2 above)
Comment (mprahl) [reply iib/web/messaging.py:277]: I'll just use `batch = requests[0].batch` instead.

File: iib/web/api_v1.py [hunk 4]
```diff
@@ -338,3 +338,60 @@ def regenerate_bundle():
...
+    :raise ValidationError: if required parameters are not supplied
+    """
+    payloads = flask.request.get_json()
+    if not isinstance(payloads, list) or not payloads:
+        raise ValidationError('The input data must be a non-empty JSON array')
+
+    batch = Batch()
+    db.session.add(batch)
+    requests = []
+    # Iterate through all the requests and verify that they are valid before committing them and
```
Comment (yashvardhannanavati) [inline iib/web/api_v1.py:359]: `# Iterate through all the requests` => `# Iterate through all the payloads`

**Comment by mprahl** [comment]
@yashvardhannanavati and @lcarva it occurred to me that I should send the state change messages before the tasks are scheduled to avoid a race condition where the workers could have started on the tasks and updated the state before all the state change messages are sent out.

I just moved it up in the last push.

**Pull request merged**